twined/schema/twine_schema.json

```diff
@@ -38,27 +38,23 @@
       "type": "object",
       "properties": {
         "datasets": {
-          "type": "array",
+          "type": "object",
           "description": "A list of entries, each describing a dataset that should be attached to / made available to the digital twin",
-          "items": {
-            "type": "object",
-            "properties": {
-              "key": {
-                "description": "A textual key identifying this dataset within the application/twin",
-                "type": "string"
-              },
-              "purpose": {
-                "description": "What data this dataset contains, eg 'the set of data files from the energy production calculation process'",
-                "type": "string",
-                "default": ""
-              },
-              "file_tags_template": {
-                "$ref": "#/$defs/file_tags_template"
+          "patternProperties": {
+            ".+": {
+              "description": "A dataset description mapped to a textual key uniquely identifying the dataset within the application/twin",
```

Comment:
> A dataset description mapped to a textual key uniquely identifying the dataset within the application/twin

Probably should say:
"A dataset representation, whose property name / key uniquely identifies the dataset to the service"